plutus-chain-index-core/src/Plutus/ChainIndex/UtxoState.hs

```diff
@@ -102,8 +127,9 @@ insert ::
        -> UtxoIndex a
        -> Either InsertUtxoFailed (InsertUtxoSuccess a)
 insert   UtxoState{_usTip=TipAtGenesis} _ = Left InsertUtxoNoTip
-insert s@UtxoState{_usTip=thisTip} ix =
-    let (before, after) = FT.split ((s <=) . snd) ix
+insert s@UtxoState{_usTip= thisTip} ix =
+    let ix'             = trimIndex 500 ix
```

Comment:
Maybe a comment about how this number will be changing in the near future?

Reply:
Is that necessary? I intend to have that PR ready by tomorrow.